test/Infrastructure/LeanCode.AuditLogs.Tests/AuditLogsFilterTests.cs

```diff
@@ -0,0 +1,175 @@
+                    .AddLeanCodeTelemetry();
+            });
+        collection.AddDbContext<TestDbContext>();
+        collection.AddTransient<IAuditLogStorage, StubAuditLogStorage>();
+        collection.AddMassTransitTestHarness(ConfigureMassTransit);
+
+        serviceProvider = collection.BuildServiceProvider();
+        harness = serviceProvider.GetRequiredService<ITestHarness>();
+    }
+
+    private static void ConfigureMassTransit(IBusRegistrationConfigurator cfg)
+    {
+        cfg.AddConsumersWithDefaultConfiguration(
+            new[] { typeof(Consumer).Assembly, typeof(AuditLogsConsumer).Assembly },
+            typeof(DefaultConsumerDefinition<>)
+        );
+
+        cfg.UsingInMemory(
+            (ctx, busCfg) =>
+            {
+                busCfg.ConfigureEndpoints(ctx, new DefaultEndpointNameFormatter("InMemory"));
+                busCfg.ConnectBusObservers(ctx);
+            }
+        );
+    }
+
+    [Fact]
+    public async Task Publishes_entity_changes_to_bus()
+    {
+        await harness.Start();
+        var consumerHarness = harness.GetConsumerHarness<Consumer>();
+
+        await harness.Bus.Publish(new TestMsg());
+        (await consumerHarness.Consumed.Any<TestMsg>()).Should().BeTrue();
+
+        harness.Published
+            .Select<AuditLogMessage>()
+            .Should()
+            .ContainSingle()
+            .Which.Context.Message.Should()
+            .BeEquivalentTo(
+                new
+                {
+                    EntityChanged = new
+                    {
+                        Ids = new string[] { SomeId },
+                        Type = typeof(TestEntity).FullName,
+                        EntityState = "Added",
+                        Changes = JsonSerializer.SerializeToDocument(TestEntity, Options),
+                    },
+                    ActionName = typeof(Consumer).FullName,
+                    ActorId = null as string,
+                    DateOccurred = Time.NowWithOffset,
+                },
+                opt => opt.ComparingByMembers<JsonElement>()
+            )
+            .And.Subject.Should()
+            .Match((s) => s.As<AuditLogMessage>().SpanId != null && s.As<AuditLogMessage>().TraceId != null);
+    }
+
+    [Fact]
+    public async Task Publishes_actor_id_when_provided_to_bus()
+    {
+        await harness.Start();
+        var consumerHarness = harness.GetConsumerHarness<Consumer>();
+
+        await harness.Bus.Publish(new TestMsg { ActorId = ActorId, });
+        (await consumerHarness.Consumed.Any<TestMsg>()).Should().BeTrue();
+
+        harness.Published
+            .Select<AuditLogMessage>()
+            .Should()
+            .ContainSingle()
+            .Which.Context.Message.Should()
+            .BeEquivalentTo(
+                new
+                {
+                    ActionName = typeof(Consumer).FullName,
+                    ActorId,
+                    DateOccurred = Time.NowWithOffset,
```

Comment:
```suggestion
                    ActorId,
```

The test suggests testing only `ActorId` so let's test only that (everything else will be tested in a different place).